### ISSUE TITLE
Stacktrace displayed for invalid JSON input

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1423]
**Version of yq**: 4.29.2
**Operating system**: linux
**Installed via**: binary release

**Describe the bug**
If the JSON input is not valid, yq exits with a stacktrace that exposes internal implementation details.

**Input JSON**
```json
{"a": 1 "b": 2}
```

**Command**
```bash
yq --input-format json'
```

**Output**
```text
panic: interface conversion: json.Token is nil, not string

goroutine 1 [running]:
github.com/mikefarah/yq/v4/pkg/yqlib.(*orderedMap).UnmarshalJSON(0x9a714d0, {0x9a176d0, 0xf, 0xf})
        /home/runner/work/yq/yq/pkg/yqlib/encoder.go:56 +0x4a6
github.com/goccy/go-json/internal/decoder.(*unmarshalJSONDecoder).DecodeStream(0x9a714e8, 0x9b1cf00, 0x0, 0x9a714d0)
        /home/runner/go/pkg/mod/github.com/goccy/go-json@v0.9.11/internal/decoder/unmarshal_json.go:63 +0x295
github.com/goccy/go-json.(*Decoder).DecodeWithOption(0x9a0f2d0, {0x8489ca0, 0x9a714d0}, {0x0, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/goccy/go-json@v0.9.11/decode.go:202 +0x111
(...)
```

**Expected behaviour**
Information why the JSON is invalid should be displayed.
[jq](https://stedolan.github.io/jq/) quits with `parse error: Expected separator between values in row 1, column 11`.
If this is not possible, suppress the stacktrace and give a generic error message, e.g. `Failed to parse JSON input`.

**Comment by mikefarah** [comment]
Will fix - thanks for raising

**Comment by mikefarah** [comment]
Fixed in 4.30.1
